inc/class-googleanalytics.php

```diff
@@ -0,0 +1,422 @@
+		$this->v3_input_legend = __( 'The Google Analytics UA ID for your network, &lsquo;UA-01234567-8&rsquo;.
+			Google will <a href=\'https://support.google.com/analytics/answer/11583528\' target=\'_blank\'>stop processing data</a>
+			for sites which use UA on July 1, 2023.', 'pressbooks' );
+		$this->v4_input_legend = __( 'The Google Analytics 4 ID for your network, e.g &lsquo;G-A123B4C5DE6&rsquo;.', 'pressbooks' );
+	}
+
+	public static function getGoogleIDSiteOption( int $version, bool $for_book ): false|string {
+		return ! $for_book ?
+			get_site_option( $version === 4 ? self::$google_id_v4_option : self::$google_id_v3_option ) :
+			get_option( $version === 4 ? self::$google_id_v4_option : self::$google_id_v3_option );
+	}
+
+	public function getInputLabel( int $version ): string {
+		return $version === 4 ? $this->v4_input_label : $this->v3_input_label;
+	}
+
+	public function getInputLegend( int $version ): string {
+		return $version === 4 ? $this->v4_input_legend : $this->v3_input_legend;
+	}
+
+	static public function init(): GoogleAnalytics {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+			self::hooks( self::$instance );
+		}
+
+		return self::$instance;
+	}
+
+	public static function hooks( GoogleAnalytics $obj ): void {
+		if ( is_admin() ) {
+			add_action( 'network_admin_menu', [ $obj, 'addNetworkMenu' ] );
+			add_action( 'admin_init', [ $obj, 'networkAnalyticsSettingsInit' ] );
+			if ( Book::isBook() && get_site_option( $obj::$is_allowed_option ) ) {
+				add_action( 'admin_menu', [ $obj, 'addBookMenu' ] );
+				add_action( 'admin_init', [ $obj, 'bookAnalyticsSettingsInit' ] );
+			}
+			add_action( 'admin_head', [ $obj, 'printScripts' ] );
+		}
+		add_action( 'wp_head', [ $obj, 'printScripts' ] );
+	}
+
+	public function addNetworkMenu(): void {
+		add_submenu_page(
+			'settings.php',
+			__( 'Google Analytics', 'pressbooks' ),
+			__( 'Google Analytics', 'pressbooks' ),
+			'manage_network_options',
+			$this->menu_slug,
+			[ $this, 'displayNetworkAnalyticsSettings' ]
+		);
+	}
+
+	public function addBookMenu(): void {
+		add_options_page(
+			__( 'Google Analytics', 'pressbooks' ),
+			__( 'Google Analytics', 'pressbooks' ),
+			'manage_options',
+			$this->menu_slug,
+			[ $this, 'displayBookAnalyticsSettings' ]
+		);
+	}
+
+	public function networkAnalyticsSettingsInit(): void {
+		$section = 'network_analytics_settings_section';
+
+		add_settings_section(
+			$section,
+			'',
+			[ $this, 'analyticsSettingsSectionCallback' ],
+			$this->network_page
+		);
+
+		$this->addNetworkSettingsFields();
+		$this->registerNetworkSettings();
+	}
+
+	private function addNetworkSettingsFields(): void {
+		$section = 'network_analytics_settings_section';
+
+		add_settings_field(
+			self::$google_id_v3_option,
+			$this->v3_input_label,
+			[ $this, 'analyticsInputCallback' ],
+			$this->network_page,
+			$section,
+			[
+				'legend' => $this->v3_input_legend,
+				'version' => 3,
```

Comment:
I'd prefer having the version number as a constant so we don't have to guess what the number is in the future when passing parameters.

```php

public const GA_VERSION_3 = 3;
public const GA_VERSION_4 = 4;

```

And then when we have to use it, we could simply use
```php
GoogleAnalytics::GA_VERSION_4;
```